sycl/source/queue.cpp

```diff
@@ -238,7 +238,13 @@ event queue::ext_oneapi_submit_barrier(const detail::code_location &CodeLoc) {
 /// group is being enqueued on.
 event queue::ext_oneapi_submit_barrier(const std::vector<event> &WaitList,
                                        const detail::code_location &CodeLoc) {
-  if (is_in_order() && WaitList.empty()) {
+  std::vector<event> FilteredWaitList;
+  std::copy_if(begin(WaitList), end(WaitList),
+               std::back_inserter(FilteredWaitList),
+               [&](const event &Event) -> bool {
+                 return detail::getSyclObjImpl(Event)->isContextInitialized();
+               });
+  if (is_in_order() && FilteredWaitList.empty()) {
```

Comment:
Do we need to create a new vector? It doesn't seem to be used more than for checking whether it's empty or not. I suggest something like:
```suggestion
  bool AnyNonEmptyEvent = std::any_of(begin(WaitList), end(WaitList),
               [&](const event &Event) -> bool {
                 return detail::getSyclObjImpl(Event)->isContextInitialized();
               });
  if (is_in_order() && AnyNonEmptyEvent) {
```

Reply:
Indeed, good point, thank you! Fixed.